common/reconciliation/common/types.go

```diff
@@ -83,16 +83,21 @@ const (
 
 // The following are types related to Invariant.
 type (
-	// Execution is an execution which should be checked or fixed.
+	// Execution is a base type for executions which should be checked or fixed.
 	Execution struct {
-		ShardID     int
-		DomainID    string
-		WorkflowID  string
-		RunID       string
+		ShardID    int
+		DomainID   string
+		WorkflowID string
+		RunID      string
+		State      int
+	}
+
+	// ConcreteExecution is a concrete execution.
+	ConcreteExecution struct {
```

Comment:
I would expect to see a CurrentExecution type defined here?

Reply:
Exactly, that will go with the next update "introduce current execution type APIs" (step 1 in the description)